File: 백준/Bronze/4153. 직각삼각형/직각삼각형.java
import java.io.*;

public class Main
{
	public static void main(String[] args) throws Exception, IOException
	{
		BufferedReader I = new BufferedReader(new InputStreamReader(System.in));
		String[] S;
		while(true)
		{
			S = I.readLine().split(" ");
			if(Integer.parseInt(S[0])==0 && Integer.parseInt(S[1])==0 && Integer.parseInt(S[2])==0)
				return;
			System.out.println(J(Integer.parseInt(S[0]),Integer.parseInt(S[1]),Integer.parseInt(S[2]))?"right":"wrong");
		}
	}
	private static boolean J(int a, int b, int c)
	{
		a*=a;
		b*=b;
		c*=c;
		if(a+b==c)
			return true;
		else if(a+c==b)
			return true;
		else if(b+c==a)
			return true;
		else
			return false;
	}
}

File: 백준/Bronze/4153. 직각삼각형/직각삼각형.cc
#include <stdio.h>

using namespace std;

bool isRightTriangle(int a, int b, int c)
{
	if (a + b == c || b + c == a || c + a == b)
		return true;
	return false;
}
int main()
{
	int N[3];

	while (true)
	{
		for (int i = 0; i < 3; i++)
		{
			scanf("%d", &N[i]);
			N[i] *= N[i];
		}
		if (N[0] && N[1] && N[2])
			printf("%s\n", isRightTriangle(N[0], N[1], N[2]) ? "right" : "wrong");
		else
			break;
	}
}